terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -460,6 +460,19 @@ func (r *ClusterResource) ModifyPlan(ctx context.Context, req resource.ModifyPla
 			"For details, see https://docs.edgeless.systems/constellation/overview/license")
 	}
 
+	// Check microservice version is set to the provider version. // must be done in modify plan to read provider data.
+	plannedVer, diags := r.getMicroserviceVersion(&plannedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if !strings.EqualFold(plannedVer.String(), r.providerData.Version.String()) {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("constellation_microservice_version"),
+			"Microservice version mismatch", fmt.Sprintf("The microservice version must be set to '%s'.", r.providerData.Version.String()),
```

Comment:
This is not a helpful error message. We should state that the Microservice version needs to equal the provider version, if this is the case. 

Something like:
> **Invalid microservice version**
> The microservice version must be set to the provider version (%s).